test/matchAnyPCat.js

```diff
@@ -0,0 +1,154 @@
+                        "cat": "x0"
+                    }, 
+                    { 
+                        "id": "c", 
+                        "cat": "x0"
+                    } 
+                ]    
+            } 
+        ]
+    };
+    var ptree1 = {
+        "id": 'root',
+        "cat": 'i',
+        "children": [
+            {
+                "id": 'a',
+                "cat": 'w',
+            },
+            {
+                "id": 'cp',
+                "cat": 'phi',
+                "children": [
+                    {
+                        "id": 'b',
+                        "cat": 'w',
+                    },
+                    {
+                        "id": 'c',
+                        "cat": 'w',
+                    }
+                ]
+            }
+        ]
+    }; 
+
+    var ptree2 = {
+        "id": 'root',
+        "cat": 'i',
+        "children": [
+            {
+                "id": 'c',
+                "cat": 'w',
+            },
+            {
+                "id": 'cp',
+                "cat": 'phi',
+                "children": [
+                    {
+                        "id": 'a',
+                        "cat": 'w',
+                    },
+                    {
+                        "id": 'b',
+                        "cat": 'w',
+                    }
+                ]
+            }
+        ]
+    }; 
+        
+    var stree2 = {
+        "id": "CP1",
+        "cat": "cp",
+        "children": [ 
+            {
+                "cat": "xp", 
+                "id": "XP_8", 
+                "children": [ 
+                    {
+                        "id": "a", 
+                        "cat": "x0"
+                    } 
+                ] 
+            },
+            { 
+                "cat": "xp", 
+                "id": "XP_9", 
+                "children": [ 
+                    {
+                        "id": "b", 
+                        "cat": "x0"
+                    } 
+                ] 
+            },
+            {
+                "cat": "xp", 
+                "id": "XP_10", 
+                "children": [ 
+                    {
+                        "id": "c", 
+                        "cat": "x0"
+                    } 
+                ] 
+            } 
+        ]
+    };
+
+    var ptree3 = {
+        "id": 'root',
+        "cat": 'i',
+        "children": [
+            {
+                "id": 'xp',
+                "cat": 'phi',
+                "children": [
+                    {
+                        "id": 'x',
+                        "cat": 'w'
+                    },
+                    {
+                        "id": 'y',
+                        "cat": 'w'
+                    },
+                    {
+                        "id": 'z',
+                        "cat": 'w'
+                    }
+                ]
+            }
+        ]
+    }; 
+
+    describe("matchAnyPCat.html", function() {
```

Comment:
Can you add useful error messages for these tests? See test/message_funcs.js for the functions and see test/matchTest.js for an example of using them.